lib/reindex/s3_bucketd.py

```diff
@@ -105,20 +105,22 @@ def __init__(self, userid=None, bucket=None, mpu=False, seekeys=False, ip="127.0
         self.total_size = 0
         if self.mpu:
             self.bucket = "mpuShadowBucket"+bucket
-        self.seekeys = seekeys
         self.ip = ip
         self.bucketd_port = bucketd_port;
 
-    def listbucket(self, session=None, marker=""):
+    def listbucket(self, session=None, marker="", versionmarker=""):
         m = marker.encode('utf8')
         mark = urllib.parse.quote_plus(m)
-        params = "%s?listingType=DelimiterMaster&maxKeys=1000&marker=%s" % (
-            self.bucket, mark)
+        v = versionmarker.encode('utf8')
+        versionmarker = urllib.parse.quote_plus(v)
+        extra = "&versionIdMarker=%s" % versionmarker
```

Comment:
From a style perspective, it feels odd to pass `versionIdMarker` as an extra param, it should be at the same level than `keyMarker` since it has the same importance. I suggest to add `&versionIdMarker...` to the format string and pass `versionmarker` in the format params.

Reply:
Good call, I've updated to that style and pushed the change.